homeassistant/components/meteoclimatic/__init__.py

```diff
@@ -0,0 +1,136 @@
+"""Support for Meteoclimatic weather data."""
+import asyncio
+import logging
+
+from meteoclimatic import MeteoclimaticClient
+from meteoclimatic.exceptions import MeteoclimaticError
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.exceptions import ConfigEntryNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    CONF_STATION_CODE,
+    DOMAIN,
+    METEOCLIMATIC_COORDINATOR,
+    METEOCLIMATIC_STATION_CODE,
+    METEOCLIMATIC_STATION_NAME,
+    METEOCLIMATIC_UPDATER,
+    PLATFORMS,
+    SCAN_INTERVAL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+STATION_CODE_SCHEMA = vol.Schema({vol.Required(CONF_STATION_CODE): cv.string})
+
+CONFIG_SCHEMA = vol.Schema(
+    {DOMAIN: vol.Schema(vol.All(cv.ensure_list, [STATION_CODE_SCHEMA]))},
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
+    """Set up Meteoclimatic weather platform."""
+
+    conf = config.get(DOMAIN)
+    if conf is None:
+        return True
+
+    for station_code_conf in conf:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=station_code_conf.copy()
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up a Meteoclimatic entry."""
+
+    station_code = entry.data[CONF_STATION_CODE]
+
+    meteoclimatic_updater = MeteoclimaticUpdater(hass, station_code)
+    await meteoclimatic_updater.async_update()
+    meteoclimatic_data = meteoclimatic_updater.get_data()
```

Comment:
It looks like this update and data check, and the coordinator refresh below are redundant.

Reply:
You are right. I just removed the coordinator refresh.